docs/DeveloperGuide.md

```diff
@@ -113,7 +113,7 @@ The `UI` component,
 
 **API** : [`Logic.java`](../src/main/java/seedu/address/logic/Logic.java)
 
-1. `Logic` uses the `Parser` class to parse the user command.
+1. `Logic` uses the `MainParser` class to parse the user command. `MainParser` relies on `Parser` of [Natty by Joe Stelmach](https://github.com/joestelmach/natty) for natural language processing.
```

Comment:
For the logic diagram, i'm sure you guys have more then add, list commands, do add in a few more too since there's still spaces and do update the image to how your logic work in ur program :)